### PR TITLE
Update Windows building instructions

### PR DESCRIPTION
**Pull request by ITAYC0HEN**
**Detailed description**
The Windows building instructions were outdated since we moved to Rizin. These changes not only make them up-to-date, but also improve them

### CODE REVIEW

File: docs/source/building.rst [hunk 1]
```diff
@@ -113,9 +113,11 @@ Requirements
...
-* A version of Visual Studio (2015, 2017 and 2019 are supported)
-* CMake
-* Qt
+* A version of `Visual Studio <https://visualstudio.microsoft.com/thank-you-downloading-visual-studio/?sku=Community&rel=16>`_ (2015, 2017 and 2019 are supported)
+* `CMake <https://cmake.org/download/>`_
+* `Qt <https://www.qt.io/download-qt-installer>`_
+* `Meson <https://mesonbuild.com/Getting-meson.html#installing-meson-with-pip>`_
+* `Ninja <https://github.com/ninja-build/ninja/releases/latest>`_
 
 Recommended Way
```
Comment (Surendrajat) [inline docs/source/building.rst:122]: Since it is the only way to build now...
```suggestion
Building Steps
```

**Review by Surendrajat — CHANGES_REQUESTED** [review]
I like the PowerShell approach. Hope we can put a simple `winget` command someday for the dependencies list :)

File: docs/source/building.rst [hunk 2]
```diff
@@ -124,19 +126,39 @@ To build Cutter on Windows machines using CMake,
...
+You can check if the binaries are available by opening PowerShell and 
+executing the following commands.
+
+.. code:: powershell
+
+   ninja --version
+   meson --version
+   cmake --version
+
+If they are not available, you can use PowerShell to add them to your path one by one:
```
Comment (Surendrajat) [inline docs/source/building.rst:138]:
```suggestion
If they are not available, you can use PowerShell to add them to your PATH one by one:
```

File: docs/source/building.rst [hunk 3]
```diff
@@ -113,9 +113,11 @@ Requirements
 Cutter works on Windows 7 or newer.
 To compile Cutter it is necessary to have the following installed:
 
-* A version of Visual Studio (2015, 2017 and 2019 are supported)
-* CMake
-* Qt
+* A version of `Visual Studio <https://visualstudio.microsoft.com/thank-you-downloading-visual-studio/?sku=Community&rel=16>`_ (2015, 2017 and 2019 are supported)
+* `CMake <https://cmake.org/download/>`_
+* `Qt <https://www.qt.io/download-qt-installer>`_
```
Comment (Surendrajat) [inline docs/source/building.rst:118]: Do we need to specify Qt5 here? I think we're not Qt6 ready yet which the installer might suggest/default to.

Comment (ITAYC0HEN) [inline docs/source/building.rst:118]:
```suggestion
* `Qt 5 <https://www.qt.io/download-qt-installer>`_
```

**Review by Surendrajat — APPROVED** [review]

File: docs/source/building.rst [hunk 4]
```diff
@@ -113,30 +113,52 @@ Requirements
...
 
-   set CMAKE_PREFIX_PATH=c:\Qt\qt-5.6.2-msvc2013-x86\5.6\msvc2013\lib\cmake
-   mkdir build
-   cd build
-   cmake-gui ..
+   # Then, add the following directory to your PATH
+   $Env:Path += ";C:\Qt\5.15.2\msvc2019_64\bin"
+
+   # Build Cutter
+   cmake -B build -DCUTTER_USE_BUNDLED_RIZIN=ON
```
Comment (yossizap) [inline docs/source/building.rst:156]: Maybe we should also include instructions for python bindings using cutter-deps? It's a basic functionality

Comment (ITAYC0HEN) [reply docs/source/building.rst:156]: I agree it is basic but I never tried it myself. Do we have any instructions for this anywhere? if no, can you draft something and push to this branch?

Comment (yossizap) [reply docs/source/building.rst:156]: I am currently having issues with windows builds on the latest commit but you can my `build.bat` which was based on the CI:

```
scripts/fetch_deps.sh # you'll need some type of bash for windows(or just manually download the zip from releases and extract) but this also replaces installing full Qt
set ARCH=x64
set CUTTER_DEPS=%CD%\cutter-deps
set PATH=%CD%\cutter-deps\qt\bin;%PATH%
call "C:\Program Files (x86)\Microsoft Visual Studio\2017\Enterprise\VC\Auxiliary\Build\vcvarsall.bat" x64
mkdir build
cd build
set PACKAGE_NAME=cutter-git-x64.Windows
cmake ^
          -DCMAKE_BUILD_TYPE=Debug ^
          -DCUTTER_USE_BUNDLED_RIZIN=ON ^
          -DCUTTER_ENABLE_PYTHON=ON ^
          -DCUTTER_ENABLE_PYTHON_BINDINGS=ON ^
          -DCUTTER_ENABLE_PACKAGING=ON ^
          -DCUTTER_PACKAGE_DEPENDENCIES=ON ^
          -DCUTTER_PACKAGE_RZ_GHIDRA=ON ^
          -DCUTTER_PACKAGE_JSDEC=ON ^
          -DCUTTER_ENABLE_DEPENDENCY_DOWNLOADS=ON ^
          -DCUTTER_ENABLE_CRASH_REPORTS=OFF ^
          -DCMAKE_PREFIX_PATH="%CUTTER_DEPS%\pyside" ^
          -DCPACK_PACKAGE_FILE_NAME=%PACKAGE_NAME% ^
          -G Ninja ^
          .
cmake --build . --config Debug
cmake --build . --config Debug --target package
```
the result is in `_CPack_Packages\win64\ZIP\cutter-git-x64.Windows`

Comment (ITAYC0HEN) [reply docs/source/building.rst:156]: I failed :<

**Pull request merged**